src/main/java/com/google/api/codegen/metacode/InitCodeContext.java

```diff
@@ -31,6 +31,10 @@
   public enum InitCodeOutputType {
     SingleObject,
     FieldList,
+    // This is used in grpc testing for when a method accepts a field list but the grpc request
+    // object is needed. This helps to compare the the request object that was constructed by the
+    // method against the request object that is initialized.
+    FieldListAndRequestObject
```

Comment:
This is because the `InitCodeOutputType` (at least for ruby) determines both the output of the initcode transformer as well as what the method definition should accept. I'm looking into splitting those two determinations into two separate enums.